updates/P6.txt

```diff
@@ -0,0 +1,135 @@
+previous block has been signed. The overall architecture remains the same.
+There are still bakers and a subset of them, the finalizers. In contrast to
+how time was split into slots in existing consensus, Concordium BFT no
+longer has a notion of slots. Instead it advances by rounds. In each round,
+a predetermined leader among the bakers should produce a block. The members
+of the finalization committee then sign this block, and their collective
+signatures are aggregated to form a quorum certificate (QC). This quorum
+certificate is then included in the next block. If the leader fails to
+produce a block in the round, or not enough signatures were gathered for a
+QC, then the finalizers will instead send timeout messages, which are
+aggregated to form a timeout certificate. Each block always contains a
+quorum certificate and may contain a timeout certificate for the previous
+round if and only if the previous round timed out. When blocks on a common
+chain in two consecutive rounds have quorum certificates, the block in the
+first of these rounds (together with its ancestors) is considered
+finalized. At this point, the protocol ensures that it cannot be rolled
+back. The two consecutive blocks must also be within the same epoch.
+
+Leaders are determined per epoch, and are known to everybody for the epoch.
+The notion of a payday, as it exists in protocols 4 and 5, remains largely
+unchanged, although payday transitions are slightly changed due to the
+technicalities of how epoch transitions occur in Concordium BFT.
+However in normal conditions there should be little observable effect on
+users of the chain.
+
+                               Changes
+
+The following behaviours are changed in protocol version 6.
+
+1. Concordium BFT takes effect. In particular this changes the format of
+   blocks, and the messages exchanged by the nodes.
+2. The election difficulty chain parameter is removed.
+3. There are new chain parameters
+   - the minimum timeout, and factors for increasing and decreasing round
+     timeout duration on successive round timeouts and finalizations,
+     respectively
+   - minimum block time, controlling minimum time between blocks in two
+     successive rounds
+   - minimum number of finalizers and maximum number of finalizers. These,
+     together with the existing finalizer relative stake threshold
+     parameter, revise the condition on being a finalizer. The relative
+     stake threshold is not interpreted relative to the total effective
+     stake of bakers. The minimum number of finalizers must be reached
+     before the relative stake threshold applies. Analogously, the number
+     of finalizers are capped.
+4. Block energy limit becomes an updatable parameter.
+5. Smart contract validation has the following changes
+   - Version 1 smart contracts can now use sign extension Wasm
+     instructions (i32.extend8_s, i32.extend16_s, i64.extend8_s,
+     i64.extend16_s, i64.extend32_s).
```

Comment:
It is not clear to me what that means. What do these extensions do?

Reply:
I added a link to the Wasm spec/instructions.

The actual sign extension proposal only has a rather poor github repository to reference really.